### PR TITLE
Fix UTs and fix core crash in DeleteFile and PutFile in case when application does not exist

### PR DESCRIPTION
**Pull request by LitvinenkoIra**
Fix UTs according to the implementation
and fix core crash in DeleteFile and PutFile in case when application does not exist

### CODE REVIEW

File: src/components/application_manager/test/commands/mobile/put_file_test.cc [hunk 1]
```diff
@@ -234,6 +234,7 @@ TEST_F(PutFileRequestTest, Run_BinaryDataGreaterThanAvaliableSpace_UNSUCCESS) {
 
   const uint32_t avaliable_space = 0u;
   EXPECT_CALL(*mock_app_, GetAvailableDiskSpace())
+      .WillOnce(Return(avaliable_space))
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/put_file_test.cc]: @LitvinenkoIra Can we use Times(n) ? Everywhere

Comment (LitvinenkoIra) [reply src/components/application_manager/test/commands/mobile/put_file_test.cc]: @AByzhynar Done 006f1d4978b7437c3146b44cb080ec716a0bdaab

File: src/components/application_manager/test/commands/mobile/put_file_test.cc [hunk 2]
```diff
@@ -293,9 +292,8 @@ TEST_F(PutFileRequestTest, Run_AddFile_SUCCESS) {
   EXPECT_CALL(*mock_app_, folder_name()).WillOnce(Return(kAppFolder));
 
   const uint32_t avaliable_space = 2u;
-  EXPECT_CALL(*mock_app_, GetAvailableDiskSpace())
-      .WillOnce(Return(avaliable_space))
-      .WillOnce(Return(avaliable_space));
+  EXPECT_CALL(*mock_app_, GetAvailableDiskSpace()).Times(3)
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/put_file_test.cc]: @LitvinenkoIra please avoid hadrcoding number of times this method will be called. Use WillRepeatedly

Comment (LitvinenkoIra) [reply src/components/application_manager/test/commands/mobile/put_file_test.cc]: @LuxoftAKutsan Done 7017728df8a6b226ee81a6e6f35268b03cbc41b9

**Comment by vkushnirenko-luxoft** [comment]
Rebuild required

**Comment by vkushnirenko-luxoft** [comment]
Rebuild required

**Pull request merged**